lib/watch-stream.js

```diff
@@ -0,0 +1,328 @@
+const { Readable } = require('streamx')
+const safetyCatch = require('safety-catch')
+
+class WatchStream extends Readable {
+  constructor (bee, opts) {
+    // no need to buffer future watches, setting hwm to 1 fixes that
+    super({ highWaterMark: 1, signal: opts.signal })
+
+    this.keyEncoding = opts.keyEncoding || bee.keyEncoding
+    this.valueEncoding = opts.valueEncoding || bee.valueEncoding
+    this.bee = bee
+    this.core = bee.core
+    this.opened = null
+
+    this.map = opts.map || null
```

Comment:
Is doing the null-checks more efficient than having a no-op function as default? Seems easier to make it no-op

Reply:
I prefer the null check here as it implies a lifecycle

Reply:
We we kill the lifecycle like we talked about, then i 100% agree